### PR TITLE
Mavlink: Fixed command sender race condition

### LINKED ISSUE #12553: Mavlink command sender race condition
While working with camera trigger via mavlink I noticed that there is a race condition occurring between mavlink instances.

When a command from PX4 needs to be sent out each mavlink instance sends the command to it's destination and then tries to add an item into the queue of the "MavlinkCommandSender".
Even though multiple instances try to insert the command into the queue, it will only be inserted once as the logic checks if the timestamp already exists.
A command item is removed from the queue as soon as the corresponding command is acknowledged by the target system which communicates through one of the mavlink instances.
If a command item remains in the queue for longer than 500ms the corresponding mavlink command is resent on ALL mavlink instances.

The race condition takes place if an acknowledge messages is received before all other mavlink instance tried to put the corresponding command into the sending queue (via MavlinkCommandSender). In that case the same command will be put into the queue again. That itself is not a problem, however, if the command is not acknowledged on those instances then the command will be re-sent after the timeout has passed to ALL other instances (including the one which already got the ack).

I have reproduced the problem as follows:
Enable camera triggering via mavlink:
TRIG_MODE 2 -> time based triggering
TRIG_INTERVAL 2000 -> send trigger every 2000ms
TRIG_INTERFACE 3 -> send trigger message via mavlink

Enable Telem2:
MAV_1_MODE -> telem2

Power the vehicle via usb or make sure that there is a mavlink instance running via usb.
This is important because we need at least two mavlink instances which send camera trigger messages.
Make sure that there is something connected to telem2 which acknowledges the camera trigger commands.
Make sure that there is nothing connected to the usb mavlink instance which acknowledges the camera trigger commands.

You will notice that the trigger messages received on TELEM2 do not come in regular intervals.

### PR DESCRIPTION
**Pull request by RomanBapst**
This is an attempt to fix #12553
See commit message for further explanation.

This fixes irregular camera trigger messages. I'm not aware if there is something I'm breaking with this change, e.g. if there was a particular reason why the current behavior is as is.

### CODE REVIEW

File: src/modules/mavlink/mavlink_command_sender.cpp [hunk 1]
```diff
@@ -172,7 +172,7 @@ void MavlinkCommandSender::check_timeout(mavlink_channel_t channel)
 			}
 		}
 
-		if (item->num_sent_per_channel[channel] < max_sent) {
+		if (item->num_sent_per_channel[channel] < max_sent && item->num_sent_per_channel[channel] != -1) {
```
Comment (julianoes) [inline src/modules/mavlink/mavlink_command_sender.cpp]: I don't exactly understand this one. Why would it still be `-1`?

**Comment by RomanBapst** [comment]
As discussed with @bkueng : If a command in the queue receives an ack, set the field "num_sent_per_channel" to -2 but leave the command in the queue. When we reach the timeout for that command we only re-send the command if "num_sent_per_channel" is not -2.

**Comment by bkueng** [comment]
Yes essentially the changes here are good (hardening the logic), but don't solve the race condition. The same command is still inserted several times. To resolve it we can wait for at least 1 timeout before removing commands from the list.

**Comment by RomanBapst** [comment]
@bkueng I updated based on your suggestion. Could you please review again. I verified basic functionality.

**Pull request merged**